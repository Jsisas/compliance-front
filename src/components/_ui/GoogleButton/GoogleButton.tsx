import * as React from 'react';
import {ReactComponent as GoogleLogo} from "../../../assets/logo/google_logo.svg";
import AlButton from "../AlButton/AlButton";

interface GoogleButtonProps {
    onClick: any;
    disabled?: boolean;
}
export function GoogleButton(props: GoogleButtonProps) {
    return (
        <AlButton onClick={props.onClick} type="secondary" disabled={props.disabled || false} style={{display: 'flex'}}><GoogleLogo style={{height: "16px", marginRight: '8px'}}/> Google</AlButton>
    );
};