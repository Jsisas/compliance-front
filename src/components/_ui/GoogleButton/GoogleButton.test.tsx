import React from 'react';
import { GoogleButton } from './GoogleButton';
import { fireEvent, render, screen } from '@testing-library/react';

describe('<GoogleButton />', () => {
	it('it matches a snapshot', () => {
		const wrapper = render(<GoogleButton onClick={() => null} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('when disabled is false then onClick should fire ', () => {
		const onClickMock = jest.fn();
		render(<GoogleButton onClick={onClickMock} disabled={false} />);

		const inputButton = screen.getByText('Google');

		fireEvent.click(inputButton);

		expect(inputButton).toBeInTheDocument();
		expect(onClickMock).toHaveBeenCalled();
	});

	it('when disabled is true then onClick should not fire ', () => {
		const onClickMock = jest.fn();
		render(<GoogleButton onClick={onClickMock} disabled={true} />);

		const inputButton = screen.getByText('Google');

		fireEvent.click(inputButton);

		expect(inputButton).toBeInTheDocument();
		expect(onClickMock).not.toHaveBeenCalled();
	});
});
