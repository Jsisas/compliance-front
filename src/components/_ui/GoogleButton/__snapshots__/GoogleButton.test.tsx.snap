// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleButton /> it renders a google button 1`] = `ShallowWrapper {}`;
